src/auth/negotiate/kerberos/negotiate_kerberos_auth.cc

```diff
@@ -370,6 +370,7 @@ main(int argc, char *const argv[])
     const unsigned char *spnegoToken = NULL;
     size_t spnegoTokenLength = 0;
 
+    default_keytab[0] = '\0';
```

Comment:
If we really want defensive programming, then we should use C++ initialization to zero the buffer (where the variable is declared):

```C++
char default_keytab[MAXPATHLEN] = "";
```

```C++
char default_keytab[MAXPATHLEN] = {};
```

The above is slightly more expensive than setting just one byte, but, in this case, all these variables are initialized just once at the program start time so we can afford that additional expense IMO. Same for other similar variables.

I do not insist on this change.

Reply:
Sounds like a good idea, and today I learn :)